probe_src/probe_py/test_probe.py

```diff
@@ -0,0 +1,253 @@
+from typer.testing import CliRunner
+import tarfile
+from .cli import app
+from . import parse_probe_log
+from . import analysis
+import pathlib
+import typing
+import networkx as nx
+import subprocess
+
+runner = CliRunner()
+
+def test_diff_cmd():
+    result = runner.invoke(app,["record","--make","diff","../flake.nix","../flake.lock"])
+    input: pathlib.Path = pathlib.Path("probe_log")
+    assert input.exists()
+    probe_log_tar_obj = tarfile.open(input, "r")
+    process_tree_prov_log = parse_probe_log.parse_probe_log_tar(probe_log_tar_obj)
+    probe_log_tar_obj.close()
+    process_graph = analysis.provlog_to_digraph(process_tree_prov_log)
```

Comment:
Can we factor out some of the code that is valid for _any_ PROBE record run to a generic run-and-check? We would still have methods like this one, but they would only have checks that are specific to this test case.

One generic check would be that successful waits correspond to a TID that is already cloned.

Reply:
Yes we can extract common checks, it will make the code less verbose. On it!

Reply:
I created a common function to test the process graph of commands that clone and open files. I extracted the common code that executes the commands and creates the process graph.

Reply:
Created a common function to test process graph of commands that clones and opens files. Extracted the common code that executes the command and creates process graph. 
 I'll add commands to ensure the code is easily understood. @charmoniumQ

Reply:
Thanks; the actual code looks good!